x/accounts/internal/implementation/implementation.go

```diff
@@ -6,14 +6,19 @@ import (
 
 	"cosmossdk.io/collections"
 	"cosmossdk.io/core/address"
+	"cosmossdk.io/core/gas"
 	"cosmossdk.io/core/header"
+
+	"github.com/cosmos/cosmos-sdk/codec"
 )
 
 // Dependencies are passed to the constructor of a smart account.
 type Dependencies struct {
 	SchemaBuilder *collections.SchemaBuilder
 	AddressCodec  address.Codec
 	HeaderService header.Service
+	GasService    gas.Service
+	StateCodec    codec.BinaryCodec
```

Comment:
do we need the codec from sdk or can we create a new one in core that only does marshal and unmarshal?

Reply:
i've adjusted the interface not to leak the whole codec functionalities